### ISSUE TITLE
Feature Request - Support Close Icon Utility

### ISSUE THREAD

**Issue by dillonredding** [reactstrap/reactstrap#1182]
Bootstrap has the [close icon utility](https://getbootstrap.com/docs/4.1/utilities/close-icon), which is just a specialized button.

It would be nice to have that supported through, perhaps, the `Button` component.

**Comment by dillonredding** [comment]
I feel like option 1 is more inline with Bootstrap's design and the approach that reactstrap takes.

If you allow for option 2, I would think that `CloseIcon` would handle the `span`/`aria-hidden` for you, so you could just do

    <CloseIcon>&times;</CloseIcon>

Perhaps `&times;` is the default, allowing you to use `<CloseIcon/>` in most cases. And maybe all that essentially translates to option 1.

**Comment by median-man** [comment]
I agree that option 1 is more consistent. I've created #1206 to add the `close` prop to `<Button />`.
